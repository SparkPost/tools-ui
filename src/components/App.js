import React, { Component } from 'react';
import { connect } from 'react-redux';
import classnames from 'classnames';
import Nav from 'components/nav/Nav';
import Footer from 'components/footer/Footer';
import { Meta, DefaultMeta } from 'components/Meta';
import { checkLogin } from 'actions/auth';
import { trackPageView } from 'actions/mixpanel';

export class App extends Component {

  // need to use this lifecycle stage to make sure that
  // the cookie is read/token in place in time
  componentWillMount() {
    this.props.checkLogin();
  }

  render() {
    const { children, location, loggedIn } = this.props;
    return (
      <div className={classnames('pageWrapper', { 'pageWrapper--loggedIn': loggedIn })}>
        <DefaultMeta />
        <Meta location={location} />
        <div className='container container--tool'>
          {children}
        </div>
        <Footer />
        <Nav location={location} />
      </div>
    );
  }
}

const mapStateToProps = ({ auth }) => ({ loggedIn: auth.loggedIn });
export default connect(mapStateToProps, { checkLogin, trackPageView })(App);
