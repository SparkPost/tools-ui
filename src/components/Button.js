import React from 'react';
import { Link } from 'react-router';
import classNames from 'classnames';
const noop = () => {};

const mapPropsToClasses = ({ type, size, accent, fullWidth, icon, states, extraClasses }) => (
  classNames('button', {
    [`button--${type}`]: type,
    [`button--${size}`]: size,
    [`button--${accent}Accent`]: accent,
    'button--full': fullWidth,
    'button--icon': icon
  }, states, extraClasses)
);

const ActionButton = (props) => {
  const { action = noop, children } = props;
  const classes = mapPropsToClasses(props);

  return (
    <button onClick={action} className={classes}>
      {children}
    </button>
  );
};

const LinkButton = (props) => {
  const { to = null, children } = props;
  const classes = mapPropsToClasses(props);

  return (
    <Link to={to} className={classes}>
      {children}
    </Link>
  );
};

export { ActionButton, LinkButton };
