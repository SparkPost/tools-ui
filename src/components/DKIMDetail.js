import React, { Component } from 'react';
import { Link } from 'react-router';
import axios from 'axios';
import config from '../config';

import HeaderTable from './DKIMDetailHeaderTable';
import SignatureTable from './DKIMSignatureTable';

export default class DKIMDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: props.email,
      error: null,
      detail: null,
      loading: false // should this just be true at first?
    }
  }

  componentDidMount() {
    this.getDetail();
  }

  getDetail() {
    const { email, detailId } = this.props;
    this.setState({ loading: true });
    return axios.get(`${config.apiBase}/messaging-tools/validations/${email}/${detailId}`)
      .then(({ data }) => {
        const { results } = data;
        this.setState({
          detail: results,
          loading: false
        });
      }, ({ response = null }) => {
        this.setState({
          error: response ? response.data.errors[0] : true,
          loading: false
        });
      });
  }

  render() {
    const { detail = {}, error } = this.state;
    return (
      <div>
        <div className='button-header'>
          <Link to={`/dkim/results/${this.props.email}`} className='button button--muted'><i className='fa fa-arrow-left'></i> Back</Link>
          <button className='button button--muted'><i className='fa fa-share-alt'></i> Share</button>
        </div>
        {error && <p>An error occurred getting details</p>}
        <HeaderTable detail={detail} />
        <SignatureTable sigs={detail && detail.sigs} />
      </div>
    )
  }
}
