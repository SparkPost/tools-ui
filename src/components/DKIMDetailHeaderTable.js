import React from 'react';

export default ({ detail }) => {
  if (!detail) {return null;}
  return (
    <table>
      <tbody>
        <tr>
          <th>Subject</th>
          <td>{detail.subject}</td>
        </tr>
        <tr>
          <th>Sender</th>
          <td>{detail.header_from}</td>
        </tr>
        <tr>
          <th>Date</th>
          <td>{new Date(detail.received).toLocaleString()}</td>
        </tr>
        <tr>
          <th>DKIM Status</th>
          <td>{detail.result ? 'Passed' : 'Failed'}</td>
        </tr>
      </tbody>
    </table>
  );
};
