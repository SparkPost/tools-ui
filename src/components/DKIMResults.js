import React, { Component } from 'react';
import { Link } from 'react-router';
import axios from 'axios';
import config from '../config';

export default class extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      results: [],
      loading: false // should this just be true at first?
    }
  }

  componentDidMount() {
    this.getResults();
  }

  getResults() {
    const { email } = this.props;
    this.setState({ loading: true });
    return axios.get(`${config.apiBase}/messaging-tools/validations/${email}`)
      .then(({ data }) => {
        const { results } = data;
        this.setState({
          results,
          loading: false
        });
      }, ({ response }) => {
        this.setState({
          error: response.data.errors[0],
          loading: false
        });
      });
  }

  renderError() {
    const { error } = this.state;
    if (!error) return null;
    return (
      <div className='error'>
        <i className='fa fa-exclamation-circle'></i>
        {error.message}
      </div>
    )
  }

  renderHeader() {
    const { email } = this.props;
    return (
      <div>
        <p>
          <strong>Results for Test Address:</strong>
          <br/>{email}
        </p>
        <button onClick={() => this.getResults()}>Refresh</button>
        <button>Share</button>
      </div>
    );
  }

  renderTable() {
    const { loading, results } = this.state;
    if (loading) return null;
    return (
      <table>
        <thead>
          <tr>
            <th>Subject</th>
            <th>DKIM Status</th>
            <th>Sender</th>
            <th>Time Delivered</th>
          </tr>
        </thead>
        <tbody>
        {
          (results.length === 0)
          ? this.renderNoMessages()
          : results.map(this.renderRow.bind(this))
        }
        </tbody>
      </table>
    );
  }

  renderNoMessages() {
    return (
      <tr className='no-messages'><td colSpan='4'>No messages have been recieved to this test address.</td></tr>
    );
  }

  renderRow({ id, subject, result, header_from, received }) {
    const { email } = this.props;
    return (
      <tr key={id}>
        <td><Link to={`/dkim/results/${email}/${id}`}>{subject}</Link></td>
        <td>{result ? 'Passed' : 'Failed'}</td>
        <td>{header_from}</td>
        <td>{new Date(received).toLocaleString()}</td>
      </tr>
    );
  }

  render() {
    const { error } = this.state;
    return (
      <div>
        {this.renderHeader()}
        {error ? this.renderError() : this.renderTable()}
      </div>
    )
  }
}
