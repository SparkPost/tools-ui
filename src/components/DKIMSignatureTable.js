import React from 'react';

const Row = ({ sig }) => {
  const { s, d, t, result } = sig;
  return (
    <tr>
      <td>{result ? 'Passed' : 'Failed'}</td>
      <td>{s}</td>
      <td>{d}</td>
      <td>{t || 'N/A'}</td>
    </tr>
  );
};

export default ({ sigs = null }) => {
  if (!sigs) {return null;}

  return (
    <table>
      <thead>
        <tr>
          <th>Status</th>
          <th>DKIM Selector (s=)</th>
          <th>Signing Domain (d=)</th>
          <th>Timestamp (t=)</th>
        </tr>
      </thead>
      <tbody>
        {sigs.map((sig) => <Row sig={sig} key={sig.s} />)}
      </tbody>
    </table>
  );
};
