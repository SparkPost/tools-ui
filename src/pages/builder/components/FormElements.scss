@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";

.input__prefix {
  position: absolute;
  top: 0;
  left: 0;

  line-height: rem(36);
  font-weight: 400;
  font-size: rem(15);
  color: grayscale(3);
  transition: 0.15s;
}

.builder-input {
  background: transparent;

  &--ghost.input__text {
    border: none;
    padding: 0;

    .has-error &, &:hover, &:active, &:focus {
      border: none;
    }

    &:hover, &:focus {
      & ~ .input__prefix {
        color: color(orange);
      }
    }
  }

  .builder-inputGroup--mx & { padding-left: rem(27); }
  .builder-inputGroup--a & { padding-left: rem(13); }
  .builder-inputGroup--include & { padding-left: rem(57); }
}
