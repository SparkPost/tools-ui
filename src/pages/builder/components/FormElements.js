import React from 'react';
import { Field } from 'redux-form';
import { ActionLink } from 'components/button/Button';
import classnames from 'classnames';

import './FormElements.scss';

export const TextInput = ({ input, prefix, extraClasses, placeholder, meta: { touched, error } }) => {
  const groupClasses = classnames('input__group', {
    [`builder-inputGroup--${prefix}`]: prefix,
    'has-error': touched && error
  });
  return (
    <div className={groupClasses}>
      <input {...input} placeholder={placeholder} type='text' className={`input__text builder-input ${extraClasses}`} autoFocus />
      {prefix && <span className='input__prefix'>{prefix}:</span>}
      {touched && (error && <span className='input__error'>{error}</span>)}
    </div>
  );
};

export const UseDefault = ({ input, domain, meta: { touched }}) => {
  const toggleText = !input.value ? 'Add' : 'Remove';
  const domainEntered = domain !== 'your domain';

  return (
    <div className='panel__body'>
      <div className='flex middle-xs'>
        <div className='col-xs-9'>
          {domainEntered && <h6 className='marginBottom--none'>{domain}</h6>}
          <p className='marginBottom--none'>Current Domain</p>
        </div>
        <div className='col-xs-3 text--right'>
          <ActionLink onClick={() => input.onChange(!input.value)}>{toggleText}</ActionLink>
          {/* <input {...input} type='checkbox'/> */}
        </div>
      </div>
    </div>
  );
};

const Host = ({ host, index, prefix, onRemove }) => (
  <div className='panel__body'>
    <div className='flex middle-xs'>
      <div className='col-xs-9'>
        <Field name={`${host}.name`} component={TextInput} prefix={prefix} extraClasses='builder-input--ghost'/>
      </div>
      <div className='col-xs-3 text--right'>
        <ActionLink onClick={onRemove}>Remove</ActionLink>
      </div>
    </div>
  </div>
);

export const Hosts = ({ fields, prefix = '', meta: { touched, error } }) => (
  <div className='panel'>
    {fields.map((host, index) => <Host key={index} host={host} prefix={prefix} index={index} onRemove={() => fields.remove(index)}/>)}
    <div className='panel__body clearfix'>
      <div className='float--right'>
        <ActionLink onClick={() => fields.push({})}>Add Host</ActionLink>
      </div>
    </div>
  </div>
);
