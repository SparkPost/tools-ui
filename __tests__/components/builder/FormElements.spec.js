import React from 'react';
import renderer from 'react-test-renderer';

import { TextInput, UseDefault, Hosts } from 'pages/builder/components/FormElements';

describe('Hosts component', () => {
  test('should render correctly with no Hosts', () => {
    expect(renderer.create(<Hosts fields={[]} />)).toMatchSnapshot();
  });
});

describe('TextInput component', () => {
  const meta = {
    touched: false,
    error: 'Error Message'
  };

  test('should render correctly with all props', () => {
    expect(renderer.create(<TextInput prefix='prefix' extraClasses='a class' placeholder='a placeholder' meta={meta} />)).toMatchSnapshot();
  });
});

describe('UseDefault component', () => {
  test('should render correctly when default turned on', () => {
    expect(renderer.create(<UseDefault input={{value: true}} domain='test.com' />)).toMatchSnapshot();
  });

  test('should render correctly when default turned off', () => {
    expect(renderer.create(<UseDefault input={{value: false}} domain='test.com' />)).toMatchSnapshot();
  });
});
