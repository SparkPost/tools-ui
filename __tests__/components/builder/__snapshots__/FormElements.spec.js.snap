exports[`TextInput component should render correctly with all props 1`] = `
<div
  className="input__group builder-inputGroup--prefix">
  <input
    autoFocus={true}
    className="input__text builder-input a class"
    placeholder="a placeholder"
    type="text" />
  <span
    className="input__prefix">
    prefix
    :
  </span>
</div>
`;
